packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,324 @@
+-   `default`: boolean - presence causes this mapping to be used when no others match the value
+
+#### Mapping element (text):
+
+_Component Name_
+
+-   `nimble-mapping-spinner`
+
+_Props/Attrs_
+
+-   `key`: string (will also have a private, typed version of this property)
+-   `label`: string - display text
+-   `default`: boolean - presence causes this mapping to be used when no others match the value
+
+### Anatomy
+
+#### `nimble-table-column-mapping`
+
+```HTML
+<template slot="${x => x.columnInternals.uniqueId}">
+    <slot
+        ${slotted('mappings')}
+        name="mapping">
+    </slot>
+    <span class="header-content">
+        <slot></slot>
+    </span>
+</template>
+```
+
+Cell view:
+
+The cell view relies on the matched mapping to provide a template to render.
+
+```HTML
+${repeat(x => (x.column as TableColumnMapping).mappings,
+    html<Mapping, TableColumnMappingCellView>`
+        ${when((x, c) => x.typedKey === (c.parent as TableColumnMappingCellView).cellRecord.value,
+            html<Mapping>`${x => x.cellViewTemplate}`
+        )}
+    `
+)}
+```
+
+Group header view:
+
+Similarly, the group header view relies on the matched mapping to provide a template to render.
+
+Note the following requires that `TableColumnMappingGroupHeaderView` has a reference to the column with which it is associated. This is needed to enumerate the column's mapping elements.
+
+```HTML
+${repeat(x => (x.column as TableColumnMapping).mappings,
+    html<Mapping, TableColumnMappingHeaderView>`
+        ${when((x, c) => x.key === (c.parent as TableColumnMappingHeaderView).groupHeaderValue,
+            html<Mapping>`${x => x.groupHeaderViewTemplate}`
+        )}
+    `
+)}
+```
+
+#### `nimble-mapping-*`
+
+```HTML
+<template slot="mapping"></template>
+```
+
+#### `nimble-mapping-icon`
+
+Cell view template:
+
+```HTML
+<${this.icon}
+    title="${x => x.label}"
+    aria-label="${x => x.label}"
+    severity="${x => x.severity}">
+</${this.icon}>
+```
+
+Group header view template:
+
+```HTML
+<${this.icon}
+    title="${x => x.label}"
+    aria-label="${x => x.label}"
+    severity="${x => x.severity}">
+</${this.icon}>
+<span
+    ${ref('span')}
+    @mouseover="${x => {
+        x.isValidContentAndHasOverflow = !!x.label && x.span!.offsetWidth < x.span!.scrollWidth;
+    }}"
+    @mouseout="${x => {
+        x.isValidContentAndHasOverflow = false;
+    }}"
+    title=${x => (x.isValidContentAndHasOverflow ? x.label : null)}>
+    ${x => x.label}
+</span>`;
+```
+
+### Grouping
+
+Grouping will be based on the record value. The grouping header will display the rendered icon/spinner/text. In the case of an icon/spinner, it will also be followed by the `label` text. This will disambiguate cases where multiple record values map to the same icon (assuming the labels are different). Text in a grouping header will be ellipsized and gain a tooltip if there is not enough room to display it all.
```

Comment:
How will group headers work for values that don't have a mapping? If a mapping is marked as `default` will its group header be used? Will all those values be in one group or in separate groups with the same header? What if no mapping is marked as `default`?

Reply:
I feel pretty strongly that we should show the raw (unmapped) data value in those cases. It seems unhelpful and confusing to show either a blank or the default mapping, specifically when there are multiple, separate, unmapped values showing the same header. I've update the spec to this effect. I realize this might be controversial, because we generally should not be mixing the display of source values and mapped values. I just think it's the more practical approach in this case.

Reply:
I am ok with this if it doesn't require us to do any contortions in the implementation to make that information available where it is needed. But thinking it through a little more, this is a bit of an error case: it indicates that the client didn't configure the column correctly for the data that it is being asked to display. Since it's related to data rather than configuration I don't think it merits triggering a validation error. But it will hopefully not be a case that users see if our clients do their jobs right.